src/main/java/com/spotify/reaper/service/SegmentRunner.java

```diff
@@ -141,7 +151,7 @@ private long getOpenFilesAmount() {
     return amountOfOpenFiles;
   }
 
-  private void runRepair() {
+  private boolean runRepair() {
```

Comment:
there's a number of `synchronized` keywords used in this class.
that makes me a little uneasy, it seems odd that it's required when it's combined with the schema now keeping running reaper instances state.